packages/eas-cli/src/build/createContext.ts

```diff
@@ -43,13 +45,48 @@ export async function createBuildContextAsync<T extends Platform>({
   const workflow = await resolveWorkflowAsync(projectDir, platform);
   const accountId = findAccountByName(user.accounts, accountName)?.id;
 
-  const credentialsCtx = new CredentialsContext({
+  let credentialsCtxOptions = {
     exp,
     nonInteractive,
     projectDir,
     user,
     env: buildProfile.env,
-  });
+  } as CredentialsContextOptions;
+  if (platform === Platform.IOS) {
+    const iosBuildProfile = buildProfile as BuildProfile<Platform.IOS>;
+    const isPartialLocalAscApiKeyProvided =
+      iosBuildProfile.ascApiKeyPath ||
+      iosBuildProfile.ascApiKeyIssuerId ||
+      iosBuildProfile.ascApiKeyId;
+    const isEntireLocalAscApiKeyProvided =
+      iosBuildProfile.ascApiKeyPath &&
+      iosBuildProfile.ascApiKeyIssuerId &&
+      iosBuildProfile.ascApiKeyId &&
+      iosBuildProfile.appleTeamId &&
+      iosBuildProfile.appleTeamType;
+
+    if (isPartialLocalAscApiKeyProvided && !isEntireLocalAscApiKeyProvided) {
+      throw new Error(
+        'To authenticate with an App Store Connect API Key, you must provide the following fields in your build profile: ascApiKeyPath, ascApiKeyIssuerId, ascApiKeyId, appleTeamId, appleTeamType.'
+      );
+    } else if (isEntireLocalAscApiKeyProvided) {
+      const keyP8 = await fs.readFile(iosBuildProfile.ascApiKeyPath!, 'utf-8');
```

Comment:
I'm not a fan of using `!`. The reason the variable is not truthy in this branch is that you can't guarantee that e.g. `.ascApiKeyPath` wasn't changed to a falsy value after the `isEntireLocalAscApiKeyProvided` was evaluated. This seems reasonable. In this case, I'd much rather prefer using nullthrows/assert

Reply:
ditto